### PR TITLE
Only remove imports when moving a type, if the imports moved to the new file.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/12730

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Tests coming.

**Comment by CyrusNajmabadi** [comment]
Tagging @dpoeschl   

David, this implements the algorithm you said you preferred.  Where the new file gets minimal usings/imports, and hte original file only removes usings/imports if it doesn't need them *and* if the new file uses them.  Imports/usings that are used by neither will be kept around in the original file.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest  please

File: src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs [hunk 1]
```diff
@@ -69,7 +71,7 @@ private class MoveTypeEditor : Editor
             /// </summary>
             /// <param name="newDocumentId">id for the new document to be added</param>
             /// <returns>the new solution which contains a new document with the type being moved</returns>
-            private async Task<Solution> AddNewDocumentWithSingleTypeDeclarationAndImportsAsync(
+            private async Task<Document> AddNewDocumentWithSingleTypeDeclarationAndImportsAsync(
```
Comment (dpoeschl) [inline src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs]: Should this be named differently now that it returns a Document? ```CreateNewDocument...``` or something?

File: src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs [hunk 2]
```diff
@@ -126,7 +130,24 @@ private class MoveTypeEditor : Editor
 
                 var updatedDocument = documentEditor.GetChangedDocument();
 
-                updatedDocument = await CleanUpDocumentAsync(updatedDocument).ConfigureAwait(false);
+                // Now, remove any imports that we no longer need *if* they were used in the new
+                // file with the moved type.  Essentially, those imports were here just to serve
+                // that new type and we shoudl remove them.  If we have *other* imports that that
```
Comment (dpoeschl) [inline src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs]: shoudl => should

**Comment by dpoeschl** [comment]
@CyrusNajmabadi Thanks for fixing this up!

**Pull request merged**